catlearn/regression/acquisition_functions.py

```diff
@@ -6,266 +6,276 @@
 from scipy.special import erf
 from scipy.stats import norm
 from collections import defaultdict, Counter
-
 from catlearn.utilities.clustering import cluster_features
 
 
-class AcquisitionFunctions(object):
-    """Base class for acquisition functions."""
-
-    def __init__(self, objective='max', k_means=3, kappa=1.5):
-        """Initialization of class.
-
-        objective : string
-            The user defines the objective of the acq. function to minimize
-            ('min') or maximize ('max').
-        k_means : int
-            Number of cluster to generate with clustering.
-        kappa: int
-            Constant that controls the explotation/exploration ratio in UCB.
-        noise: float
-            Small number must be added in the denominator for stability.
-        """
-        self.objective = objective
-        self.k_means = k_means
-        self.noise = 1e-6
-        self.kappa = kappa
-
-    def rank(self, targets, predictions, uncertainty, train_features=None,
-             test_features=None, metrics=['cdf', 'optimistic', 'UCB', 'EI',
-                                          'PI']):
-        """Rank predictions based on acquisition function.
-
-        Parameters
-        ----------
-        targets : list
-            List of known target values.
-        predictions : list
-            List of predictions from the GP.
-        uncertainty : list
-            List of variance on the GP predictions.
-        train_features : array
-            Feature matrix for the training data.
-        test_features : array
-            Feature matrix for the test data.
-        metrics : list
-            list of strings.
-            Accepted values are 'cdf', 'UCB', 'EI', 'PI', 'optimistic' and
-            'gaussian'.
-
-        Returns
-        -------
-        res : dict
-            A dictionary of lists containg the fitness of each test point for
-            the different acquisition functions.
-
-        """
-        self.targets = targets
-        self.predictions = predictions
-        self.uncertainty = uncertainty
-        self.train_features = train_features
-        self.test_features = test_features
-
-        # Create dictionary for each acquisition function.
-        res = {}
+def cdf(y_best, predictions, uncertainty):
```

Comment:
I think this is repeated further down with a more conventional name. @jagarridotorres is this correct? If so it can just be removed as a function.

Reply:
@pcjennings I have never used cdf, is it the same than PI?

Reply:
@jagarridotorres yeah, the math is pretty much the same right? We just called it this before we were aware that other people called it PI.